R/model-tools.R

```diff
@@ -139,14 +143,18 @@ enw_priors_as_data_list <- function(priors) {
 #' fit_priors
 #'
 #' enw_replace_priors(default_priors, fit_priors)
-enw_replace_priors <- function(priors, custom_priors) {
-  custom_priors <- coerce_dt(custom_priors)[
+enw_replace_priors <- function(priors, custom_priors, copy = TRUE) {
+  custom_priors <- coerce_dt(
+    custom_priors, select = c("variable", "mean", "sd"), copy = copy
```

Comment:
select and so copy so no arg

Reply:
the full body here is:
```r
enw_replace_priors <- function(priors, custom_priors) {
  custom_priors <- coerce_dt(
    custom_priors, select = c("variable", "mean", "sd")
  )[
    ,
    .(variable = gsub("\\[([^]]*)\\]", "", variable),
      mean = as.numeric(mean), sd = as.numeric(sd))
  ]
  variables <- custom_priors$variable
  priors <- coerce_dt(
    priors, required_cols = "variable", copy = copy
  )[!(variable %in% variables)]
  priors <- rbind(priors, custom_priors, fill = TRUE)
  return(priors[])
}
```

that suggests that `priors` might not have mean, sd OR might have extra columns (b/c `fill = TRUE`) - accurate?

or should we be using the `select = ...` argument in the second `coerce_dt(priors, ...)`?

Reply:
(also: this might be best captured as part of some optimization-pass or leverage-coerce_dt issue(s))

Reply:
agree